yt_dlp/extractor/youtube.py

```diff
@@ -4080,50 +4061,15 @@ def _reload_with_unavailable_videos(self, item_id, data, ytcfg):
             note='Downloading API JSON with unavailable videos')
 
     def _extract_webpage(self, url, item_id, fatal=True):
-        retries = self.get_param('extractor_retries', 3)
-        count = -1
-        webpage = data = last_error = None
-        while count < retries:
-            count += 1
-            # Sometimes youtube returns a webpage with incomplete ytInitialData
-            # See: https://github.com/yt-dlp/yt-dlp/issues/116
-            if last_error:
-                self.report_warning('%s. Retrying ...' % last_error)
-            try:
-                webpage = self._download_webpage(
-                    url, item_id,
-                    note='Downloading webpage%s' % (' (retry #%d)' % count if count else '',))
-                data = self.extract_yt_initial_data(item_id, webpage or '', fatal=fatal) or {}
-            except ExtractorError as e:
-                if isinstance(e.cause, network_exceptions):
-                    if not isinstance(e.cause, compat_HTTPError) or e.cause.code not in (403, 429):
-                        last_error = error_to_compat_str(e.cause or e.msg)
-                        if count < retries:
-                            continue
-                if fatal:
-                    raise
-                self.report_warning(error_to_compat_str(e))
-                break
-            else:
-                try:
-                    self._extract_and_report_alerts(data)
-                except ExtractorError as e:
-                    if fatal:
-                        raise
-                    self.report_warning(error_to_compat_str(e))
-                    break
+        extract_yt_initial_data = self._handle_incomplete_data(self.extract_yt_initial_data, ('contents', 'currentVideoEndpoint'))
 
-                if dict_get(data, ('contents', 'currentVideoEndpoint')):
-                    break
-
-                last_error = 'Incomplete yt initial data received'
-                if count >= retries:
-                    if fatal:
-                        raise ExtractorError(last_error)
-                    self.report_warning(last_error)
-                    break
+        @self._retry(self._retry_on_error, fatal)
+        def extract_webpage_and_data(url, item_id, *args, **kwargs):
+            webpage = self._download_webpage(url, item_id, *args, **kwargs)
+            data = extract_yt_initial_data(item_id, webpage or '', fatal=fatal)
```

Comment:
if `webpage` is None (and/or we fail to extract yt initial data), `self._handle_incomplete_data` crashes yt-dlp